test/sets.jl

```diff
@@ -344,3 +344,37 @@ end
     @test typeof(cssset) == Set{String}
     @test cssset == Set(["foo", "bar"])
 end
+
+@testset "replace! & replace" begin
+    maybe1(v, p) = if p Some(v) end
```

Comment:
The point of using `Some` is to be able to distinguish `nothing`. Can you add a test with `Some(nothing)` to ensure this works?